node/src/components/consensus/era_supervisor.rs

```diff
@@ -1224,3 +1227,18 @@ fn instance_id(protocol_config: &ProtocolConfig, era_id: EraId) -> Digest {
     });
     result.into()
 }
+
+/// The number of past eras whose validators are still bonded. After this many eras, a former
+/// validator is allowed to withdraw their stake, so their signature can't be trusted anymore.
+///
+/// A node keeps `2 * bonded_eras` past eras around, because the oldest bonded era could still
+/// receive blocks that refer to `bonded_eras` before that.
+fn bonded_eras(protocol_config: &ProtocolConfig) -> u64 {
+    protocol_config.unbonding_delay - protocol_config.auction_delay
+}
+
+/// The oldest era whose validators are still bonded.
+// This is public because it's used in reactor::validator::tests.
+pub(crate) fn oldest_bonded_era(protocol_config: &ProtocolConfig, current_era: EraId) -> EraId {
+    current_era.saturating_sub(bonded_eras(protocol_config))
+}
```

Comment:
I wonder if this should take the last activation point into account, too...

Reply:
Good point, I'll add that.
(It doesn't matter in the places where `oldest_bonded_era` is currently called, but it's still more correct.)

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/1052/commits/96959a82b221ea4e2b3996c49ddbe1bb99644968.